spirv_glsl.cpp

```diff
@@ -804,6 +804,9 @@ bool CompilerGLSL::ssbo_is_std430_packing(const SPIRType &type)
 
 string CompilerGLSL::layout_for_variable(const SPIRVariable &var)
 {
+	if (is_legacy_es() || (!options.es && options.version < 330))
```

Comment:
Ye, that makes sense, the 420 define is already added as a conditional define, so it's kinda ugly. I suppose the check could be. If 420 is enabled (default), the legacy check is done, if not, check for core support.

One problem really is that layout() on uniform buffers cannot be used unless 420 is enabled, so this check must be quite variable in what it will pass.